spec/order_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'order' 
+describe Order do
+  default_menu = Menu::MENU
+  let(:menu) { instance_double(Menu, dishes: default_menu) }
+  subject {described_class.new(menu)}
+  let(:dish) {"dal"}
+  let(:quantity) {2}
+  let(:price) {6.50}
+
+  describe "#add" do 
+  before do 
+    subject.add(dish, quantity)
+  end
+
+    it 'adds dish name, price and quantity to basket' do
+      expect(subject.basket).to include({name: dish, item_price: price, quantity: quantity, total_cost: quantity * price})
```

Comment:
I mentioned testing behaviour over state before, but this is an example of testing state. You already have a nice `#view_order` method which displays the order in a way you can control, so you don't want to expose the `#basket` to the outside world. Better to test the _outputs_ of the `#add` method than what it does to the internal state.

If there are no outputs, that's fine, you can use another method to check the result and test the two together.